aten/src/ATen/native/Sorting.cpp

```diff
@@ -226,9 +226,8 @@ Tensor quantile_compute(
   // NOTE: this check is only performed when running on the CPU to avoid
   // synchronizing an accelerator with the CPU
   if (self.device().is_cpu()) {
-    TORCH_CHECK(
-        q.ge(0).logical_and_(q.le(1)).all().item<bool>(),
-        "quantile() q values must be in the range [0, 1]");
+    auto q_in_range = q.ge(0).logical_and_(q.le(1));
+    at::assert_all_true(q_in_range, "quantile() q values must be in the range [0, 1]");
```

Comment:
@kshitij12345 -- Instead of `assert_all_true`, we could write this as
```
TORCH_CHECK(
    torch.equal(q_in_range.all(), at::tensor(true)),
    quantile() q values must be in the range [0, 1]);
```

Reply:
Nice! Thanks for the suggestion!